test/docfx.Test/DocfxLanguageServerTestHost.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using OmniSharp.Extensions.JsonRpc.Testing;
+using OmniSharp.Extensions.LanguageProtocol.Testing;
+using OmniSharp.Extensions.LanguageServer.Client;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.Docs.Build
+{
+    public class DocfxLanguageServerTestHost : LanguageServerTestBase
+    {
+        private readonly Channel<LanguageServerNotification> _notificationsChannel =
+           Channel.CreateUnbounded<LanguageServerNotification>();
+
+        private Task startUpTask;
+
+        private static readonly string[] s_notificationsToListen = { "window/showMessage" };
+
+        protected ILanguageClient Client { get; private set; }
+
+        public DocfxLanguageServerTestHost()
+            : base(new JsonRpcTestOptions())
+        {
+        }
+
+        public async Task InitializeAsync()
+        {
+            Client = await InitializeClient(x =>
+            {
+                x.WithCapability(new WorkspaceEditCapability()
+                {
+                    DocumentChanges = true,
+                    FailureHandling = FailureHandlingKind.Undo,
+                });
+
+                foreach (var name in s_notificationsToListen)
+                {
+                    x.OnJsonNotification(name, @params =>
+                    {
+                        _notificationsChannel.Writer.TryWrite(new LanguageServerNotification(name, @params));
+                    });
+                }
+            });
+
+            await startUpTask;
+        }
+
+        public void SendNotification(LanguageServerNotification notification)
+        {
+            Client.SendNotification(notification.Method, notification.Params);
+        }
+
+        public async Task<LanguageServerNotification> GetExpectedNotification(string method)
+        {
+            try
+            {
+                using var cts = new CancellationTokenSource(10000);
+                while (true)
+                {
+                    var notification = await _notificationsChannel.Reader.ReadAsync(cts.Token).AsTask();
+                    if (notification.Method.Equals(method, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return notification;
+                    }
+                }
+            }
+            catch (OperationCanceledException)
```

Comment:
consider don't catch the exception and let the method throw to fail the test, to tell the different between not receiving such notification and received a notification but the params is `null`